Src/Newtonsoft.Json/Serialization/JsonSerializerInternalWriter.cs

```diff
@@ -483,39 +483,36 @@ private void SerializeObject(JsonWriter writer, object value, JsonObjectContract
                 }
             }
 
-            if (contract.ExtensionDataGetter != null)
+            IEnumerable<KeyValuePair<object, object>> extensionData = contract.ExtensionDataGetter?.Invoke(value);
+            if (extensionData != null)
             {
-                IEnumerable<KeyValuePair<object, object>> extensionData = contract.ExtensionDataGetter(value);
-                if (extensionData != null)
+                foreach (KeyValuePair<object, object> e in extensionData)
                 {
-                    foreach (KeyValuePair<object, object> e in extensionData)
-                    {
-                        JsonContract keyContract = GetContractSafe(e.Key);
-                        JsonContract valueContract = GetContractSafe(e.Value);
+                    JsonContract keyContract = GetContractSafe(e.Key);
+                    JsonContract valueContract = GetContractSafe(e.Value);
 
-                        bool escape;
-                        string propertyName = GetPropertyName(writer, e.Key, keyContract, out escape);
+                    bool escape;
+                    string propertyName = GetPropertyName(writer, e.Key, keyContract, out escape);
 
-                        propertyName = (contract.ExtensionDataNameResolver != null)
-                            ? contract.ExtensionDataNameResolver(propertyName)
-                            : propertyName;
+                    propertyName = (contract.ExtensionDataNameResolver != null)
```

Comment:
If I'm not mistaken, this could be further reduced to:

```c#
propertyName = contract.ExtensionDataNameResolver?.Invoke(propertyName) ?? propertyName;
```

But since `propertyName` doesn't always change, perhaps the extra assignment could be avoided with a regular `if` block:

```c#
if (contract.ExtensionDataNameResolver != null)
{
    propertyName = contract.ExtensionDataNameResolver(propertyName);
}
```